### DISCUSSION TITLE
Can I utilize gpu to build cpu index? Or can I convert cpu/gpu index to the other type?

### DISCUSSION THREAD

**Discussion by Wiley-Z** [milvus-io/milvus#34846] [Q&A and General discussion] [answered]
I've assigned a gpu for milvus-standalone. It seems not using gpu when creating cpu index, so it is rather slow for ivf_pq index. Can I modify some configs to utilize gpu for cpu index building? Or can I build an gpu index and convert it to the corresponding cpu index type?

**Comment by xiaofan-luan** [comment]
not for now. but this is ideally doable for IVF index, and might be doable for Cagra soon as well.

**Comment by xiaofan-luan** [comment]
@liliu-z @Presburger 

let me know if there is any progress

**Comment by presburger** [comment, marked as answer]
@Wiley-Z  Currently, Milvus does not support using GPUs to build CPU indexes. In the next version of Milvus, it will support building indexes with GPUs for Cagra indexing while using the CPU for searching. We will update the documentation accordingly.

**Reply by xiaofan-luan** [reply]
ivf should ideally be also work right?
